### ISSUE TITLE
gatsby-plugin-use-query-params jumps to top of the page when a query param is set

### ISSUE THREAD

**Issue by alexhillel** [alexluong/gatsby-packages#33]
Excellent work wrapping this hook up for use in Gatsby, but is there a reason there is a jump to the top of the document when a query parameter is set? This is also reported by various other people in https://github.com/pbeshai/use-query-params/issues/56

Is there a way for this to be solved in the plugin?

**Comment by Morgantheplant** [comment]
Running into this problem as well. I am very new to Gatsby so bear with me.. 
Here's what I found so far. Added this line above the component that's using useQueryParam:
```

if(window !== undefined) {
  window.scrollTo = ()=> { 
    debugger
  }
}
```
Sure enough that was being called and I used that to walk up the call stack. It's a little hard to follow since this bounces between Gatsby and the `scroll-behavior` library via callbacks.

I could be wrong, but it appears like the issue might be on this line: 
https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/navigation.js#L144

```
  if (prevRouterProps) {
    const {
      location: { pathname: oldPathname },
    } = prevRouterProps
    if (oldPathname === pathname) {
      // Scroll to element if it exists, if it doesn't, or no hash is provided,
      // scroll to top.
      return hash ? decodeURI(hash.slice(1)) : [0, 0]
    }
  }
```  

Looks like since the old path matches the next path the current target is set to [0,0] (top of the screen). Since my path is the same and just the query param is changing i think this might be it. 

On [the lines above](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/navigation.js#L127-L138) it looks like you can override this via the APIrunner. 

so [following the directions here](https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll) in the `gatsby-browser.js` module I added the following lines to igrnore the affected routes:

```
export const shouldUpdateScroll = ({
  routerProps: { location },
  getSavedScrollPosition,
}) => {
  // hack to override `shouldUpdateScroll` behavior to bypass useQueryParams in product-checkout
  if (location.search.includes("step=")) {
    return false;
  }
  const currentPosition = getSavedScrollPosition(location);
  return currentPosition || [0, 0];
};
```
Still testing this out, but figured I might share my findings so far. Hopefully that gives you something to work with. Let me know if you discover anything

**Comment by AndiZandi** [comment]
Worked perfectly! Thanks so much!!! Saved my life :)

It'll still scroll up when you have no params anymore, so if you remove all params via buttons or so, itll scroll to the top. Can bypass this as well by not only checking for the param and returning false in the case of its existence, but also checking for location.pathname or whatever you want.
